config/source.go

```diff
@@ -0,0 +1,58 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"strconv"
+
+	"github.com/conduitio/conduit-connector-gcp-pubsub/config/validator"
+	"github.com/conduitio/conduit-connector-gcp-pubsub/models"
+)
+
+// A Source represents the source configuration needed for GCP Pub/Sub.
+type Source struct {
+	general
+	SubscriptionID string `validate:"omitempty,object_name"`
```

Comment:
Why it is separate from general?

Reply:
It would be great to extend the `general` struct of configuration fields with specific fields for `Source` and `Destination`.